### PR TITLE
HdSpace: Add IMDB support.

### PR DESCRIPTION
**Pull request by cadatoiva**
Related #4859

### CODE REVIEW

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 1]
```diff
@@ -41,6 +41,8 @@ public HDSpace(IIndexerConfigurationService configService, WebClient wc, Logger
             Encoding = Encoding.UTF8;
             Language = "en-us";
             Type = "private";
+            TorznabCaps.SupportsImdbMovieSearch = true;
+            TorznabCaps.SupportsImdbTVSearch = true;
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: @garfield69 In other trackers we have only `SupportsImdbMovieSearch` and it works for both, movies and tv. What is the right way? What torznab says about?

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 2]
```diff
@@ -99,13 +101,19 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                queryCollection.Add("search", searchString);
+            if (query.IsImdbQuery)
+            {
+                queryCollection.Add("options", "2");
+                queryCollection.Add("search", query.ImdbIDShort);
+            }
+            else
+            {
+                queryCollection.Add("options", "0");
+                queryCollection.Add("search", query.GetQueryString().Trim());
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: I think you don't need to trim.

File: src/Jackett.Common/Indexers/HDSpace.cs (hunk 1 above)
Comment (garfield69) [reply src/Jackett.Common/Indexers/HDSpace.cs]: >In other trackers we have only SupportsImdbMovieSearch and it works for both, movies and tv. What is the right way? What torznab says about?

Originally IMDB was only a Movies site, and they subsequently added TV later.
So the Torznab specs began with just the movie-search accepting imdbid.
Sonarr v2 only ever provided season,ep for their tv-search Torznab queries.

The use of SupportsImdbMovieSearch came about when Sonarr v3 added imdbid to their tv-search, we discovered that some sites did not support using imdbid for tv but just for movies (RARBG).

Much later we found a site that did support imdbid for tv, (TVStore) and so SupportsImdbTVSearch  was added too.

When SupportsImdbMovieSearch support was added, (during the period when was lots of development members on Jackett) the team did go around checking and adding it to some of the c# indexers.
Bu when SupportsImdbTVSearch  support was added (during the time when the team was down to just a few members), only TVStore was updated. And this is the only reason why you don't see it in more places.

So now we are at the point where we probably should kick off a survey of all the C# indexers  and add SupportsImdbTVSearch to them if they too support imdbid for tv-search.

Comment (ngosang) [reply src/Jackett.Common/Indexers/HDSpace.cs]: I've been thinking for a while that we need to create a matrix/table with all the trackers and all the supported functionalities (search, categories, banner, torrent, magnet, seed, pers, leecher, grabs, imdb search, imdb links, ...). It seems the time has come. Each functionality should have 3 states: implemented, todo, not supported. I don't know what is best, the wiki or a Google Excel..

Comment (garfield69) [reply src/Jackett.Common/Indexers/HDSpace.cs]: spreadsheet is the superior option as demonstrated by the maintainers list ;-)

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/HDSpace.cs]: This matrix should also become a basis for setting up new trackers as they come in, as a checklist of things to look for and implement.

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/HDSpace.cs]: I'll build up another spreadsheet to put in the maintainer's chat.

**Pull request merged**